File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
      map<int,int> sum;
      int n = nums.size();
      for(int i=0;i<n;i++){
        if(sum.find(target-nums[i]) != sum.end()){
            return {sum[target-nums[i]],i};
        }
        sum[nums[i]] = i;
      }
      return {};
    }
};

File: Data Structures & Algorithms/duplicate-integer/submission-1.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        mp = {}
        for n in nums:
            mp[n] = mp.get(n, 0) + 1
        for key in mp:
            if(mp[key]>1):
                return True
        return False